### PR TITLE
Safe Buffer, revert breaking change, use md5.js, less sarcasm

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

**Comment by dcousens** [comment]
@calvinmetcalf ready to merge,  reverts to `1.0.0` API with hard-coded use of `md5.js`... this should drop in and work.

I don't think the `createHash` function provisor is required.

The OpenSSL tests and internal refactor were great though,  thanks @fanatid :heart: .

File: test/index.js [hunk 1]
```diff
@@ -6,44 +6,37 @@ try {
...
-function createHashMD5 () { return crypto.createHash('md5') }
 var keyLen = 32
 var ivLen = 16
 
-var counts = [1, 2, 100]
-for (var i = 0; i < counts.length; ++i) {
-  var count = counts[i]
+for (var i = 0; i < 1000; ++i) {
+  var password = crypto.randomBytes(10 + Math.round(Math.random() * 100))
+  var salt = crypto.randomBytes(8)
```
Comment (fanatid) [inline test/index.js]: I think this should be repeatable otherwise when test fails sometime password/salt will not be known
(can be done with xorshift for example)

File: README.md [hunk 2]
```diff
@@ -1,59 +1,51 @@
...
 ## API
+`EVP_BytesToKey(password, salt, keyLen, ivLen)`
 
-`EVP_BytesToKey(createHash, salt, data, count, keyLen, ivLen)`
-
-* `createHash` - function which return [node Hash class][2].
+* `password` - `Buffer`, password used to derive the key data.
 * `salt` - 8 byte `Buffer` or `null`, salt is used as a salt in the derivation.
-* `data` - `Buffer`, data which is used to derive the keying data.
-* `count` - `number`, count is the iteration count to use.
```
Comment (fanatid) [inline README.md]: OOI why `count` is removed?

File: test/index.js (hunk 1 above)
Comment (calvinmetcalf) [inline test/index.js]: the 'fuzzing' style tests can be pretty helpful to find weird edge cases, that being said I'm not against also doing some deterministic tests as well

File: README.md (hunk 2 above)
Comment (fanatid) [inline README.md]: oh, you're right
I thought that `count` was in 1.0.0

**Comment by dcousens** [comment]
@calvinmetcalf I can't publish,  pushed up `v1.0.1`,  but you'll have to publish.

**Comment by calvinmetcalf** [comment]
you should be an owner now

**Comment by dcousens** [comment]
Published!
